pyqg/xarray_output.py

```diff
@@ -0,0 +1,152 @@
+        'filterfac': (),
+        'twrite': (),
+        'tmax': (),
+        'tavestart': (),
+        'tsnapstart': (),
+        'taveint': (),
+        'tsnapint': (),
+        'ntd': (),
+        'pmodes': (),
+        'radii': (),
+    }
+
+    # dict for coordinate attributes 
+    coord_attr_database = {
+        'time': {'long_name': 'model time', 'units': 'seconds',},
+        'z': {'long_name': 'vertical levels', 'units': 'none',},
+        'x': {'long_name': 'real space grid points in the x direction', 'units': 'grid point',},
+        'y': {'long_name': 'real space grid points in the y direction', 'units': 'grid point',},
+        'l': {'long_name': 'spectal space grid points in the l direction', 'units': 'meridional wavenumber',},
+        'k': {'long_name': 'spectal space grid points in the k direction', 'units': 'zonal wavenumber',},
+        'nx': {'long_name': 'number of real space grid points in x direction', 'units': 'none',},
+        'ny': {'long_name': 'number of real space grid points in y direction (default: nx)', 'units': 'none',},
+        'nz': {'long_name': 'number of vertical levels', 'units': 'none',},
+        'nl': {'long_name': 'number of spectral space grid points in l direction', 'units': 'grid point',},
+        'nk': {'long_name': 'number of spectral space grid points in k direction', 'units': 'grid point',},
+        'rek': {'long_name': 'linear drag in lower layer', 'units': 'seconds\u207B\u00B9',},
+        'tc': {'long_name': 'model timestep', 'units': 'seconds',},
+        'dt': {'long_name': 'numerical timestep', 'units': 'seconds',},
+        'L': {'long_name': 'domain length in x direction', 'units': 'meters',},
+        'W': {'long_name': 'domain length in y direction', 'units': 'meters',},
+        'filterfac': {'long_name': 'amplitude of spectral spherical filter', 'units': '',},
+        'twrite': {'long_name': 'interval for cfl writeout', 'units': 'number of timesteps',},
+        'tmax': {'long_name': 'total time of integration', 'units': 'seconds',},
+        'tavestart': {'long_name': 'start time for averaging', 'units': 'seconds',},
+        'tsnapstart': {'long_name': 'start time for snapshot writeout', 'units': 'seconds'},
+        'taveint': {'long_name': 'time interval for accumulation of diagnostic averages', 'units': 'seconds'},
+        'tsnapint': {'long_name': 'time interval for snapshots', 'units': 'seconds',},
+        'ntd': {'long_name': 'number of threads used', 'units': 'none',},
+        'pmodes': {'long_name': 'vertical pressure modes', 'units': 'none',},
+        'radii': {'long_name': 'deformation radii', 'units': 'meters',},
+    }
+
+    # dict for dataset attributes
+    ds_attr_database = {
+        'title': 'pyqg: Python Quasigeostrophic Model',
+        'institution': '',
+        'source': ('version: {}'.format(pyqg.__version__)),
+        'history': '',
+        'references': 'https://pyqg.readthedocs.io/en/latest/index.html',
+        'comment': '', 
+    }
```

Comment:
All of these databases can me moved _outside_ the function definition. (They are essentially "constants" which don't depend on the input.) This will help keep the function itself more concise.